lib/Analysis/Allocation.cpp

```diff
@@ -127,9 +157,17 @@ class AllocationAnalysis {
       // TODO(Keren): Reduce with index is not supported yet.
       auto value = op->getOperand(0);
       if (auto tensorType = value.getType().dyn_cast<RankedTensorType>()) {
-        auto bytes = tensorType.getNumElements() *
-                     tensorType.getElementTypeBitWidth() / 8;
-        allocation->addBuffer<BufferT::BufferKind::Scratch>(op, bytes);
+        if (auto layout =
```

Comment:
use value.getType().isa<RankedTensorType>() instead of dyn_cast since layout is not used hereafter

Reply:
Thanks, done.